celery/backends/database/session.py

```diff
@@ -4,8 +4,7 @@
 from kombu.utils.compat import register_after_fork
 from sqlalchemy import create_engine
 from sqlalchemy.exc import DatabaseError
-from sqlalchemy.ext.declarative import declarative_base
-from sqlalchemy.orm import sessionmaker
+from sqlalchemy.orm import sessionmaker, declarative_base
```

Comment:
Why should we drop support for older versions?

Reply:
declarative_base() - this is the same declarative base class that has been in use throughout the span of the declarative system, except it now references a registry object internally and is implemented by the registry.generate_base() method which can be invoked from a registry directly. The declarative_base() function creates this registry automatically so there is no impact on existing code. The sqlalchemy.ext.declarative.declarative_base name is still present, emitting a 2.0 deprecation warning when 2.0 deprecations mode is enabled.

Reply:
this is not a so big change from a sqla users perspective. as we are just updating the code path, nothing functional is changing with this and not in database level.

Reply:
I understand. But why should we drop support for older versions?

Reply:
sqla 1.4 is much slimmer in API & lay the groundwork for sqla 2.0 which will be python 3 only. if we are going to support multiple versions support, that won;t be hard as its just import and fallback to another version if import fail.

Reply:
I think we should support multiple versions until the next major version of Kombu.
It should be easy to do so.
Please add a TODO so we don't forget to remove the fallback once we drop support for sqlalchemy<1.4.

Reply:
ok